Sources/ForageSDK/Foundation/Telemetry/ForageLogger.swift

```diff
@@ -0,0 +1,102 @@
+//
+//  File.swift
+//
+//
+//  Created by Danilo Joksimovic on 2023-07-26.
+//
+
+import Datadog
+import Foundation
+
+// TODO: move this?
+protocol ForageLogger {
+    func setCustomerID(_ customerID: String)
+    func setMerchantAccount(_ merchantAccount: String)
+    func initialize(_ environment: EnvironmentTarget)
+
+    func notice(_ message: String, attributes: [String: Encodable]?)
+    func info(_ message: String, attributes: [String: Encodable]?)
+    func warn(_ message: String, error: Error?, attributes: [String: Encodable]?)
+    func error(_ message: String, error: Error?, attributes: [String: Encodable]?)
+    func critical(_ message: String, error: Error?, attributes: [String: Encodable]?)
+}
+
+public class DatadogLogger: ForageLogger {
+    static let shared = DatadogLogger()
+    private static let DD_CLIENT_TOKEN: String = "pub1e4572ba0f5e53df108c333d5ec66c02"
+    private static var logger: Logger? = nil
+
+    private init() {}
+
+    internal func initialize(_ environment: EnvironmentTarget) {
+        if DatadogLogger.logger != nil {
+            return
+        }
+
+        // TODO: only initialize if not running locally?
+
+        Datadog.initialize(
+            appContext: .init(),
+            trackingConsent: .granted,
+            configuration: Datadog.Configuration
+                .builderUsing(clientToken: DatadogLogger.DD_CLIENT_TOKEN, environment: environment.rawValue)
+                .set(serviceName: "ios-sdk")
+                .set(endpoint: .us1)
+                .build()
+        )
+
+        Datadog.verbosityLevel = .info
```

Comment:
Curious why we want to do the initialization of the Datadog singleton inside the instance method of `initialize` instead of somewhere more upstream in the application? From what I can tell, this initialization-related code is not something that `DatadogLogger` cares about. Instead, `DatadogLogger` seems only to care that `Datadog` was indeed initialized **before** an instance of `DatadogLogger` is created. 

What do you think about a pattern like:
```swift
// somewhere in the app initialization
Datadog.initialize(
    appContext: .init(),
    trackingConsent: .granted,
    configuration: Datadog.Configuration
        .builderUsing(clientToken: DatadogLogger.DD_CLIENT_TOKEN, environment: environment.rawValue)
        .set(serviceName: "ios-sdk")
        .set(endpoint: .us1)
        .build()
)
                
// in ForageLogger.swift
public class DatadogLogger: ForageLogger {
    // not shared since its not static
    private var logger: Logger? = nil

    private init(...loggerRelatedConfig) {
        this.logger = Logger.builder
            // apply logerRelatedConfig
            .build()
    }

    // no more initialize function since all of the async setup is done before
}
```